client/chat/channel-info-card.tsx

```diff
@@ -105,29 +162,22 @@ export function ConnectedChannelInfoCard({
   const isUserBanned =
     isFetchError(joinChannelError) && joinChannelError.code === ChatServiceErrorCode.UserBanned
 
-  const icon =
-    channelInfo && (!channelInfo.private || isUserInChannel) ? (
-      <StyledChannelIcon as={ChannelIcon} />
-    ) : (
-      <ErrorChannelIcon as={WarningIcon} />
-    )
-
-  let subtitle
+  let descriptionText
   if (isChannelNotFound) {
-    subtitle = (
+    descriptionText = (
       <ErrorText>
         This channel could not be found. It might not exist, or it may have been re-created by
         someone else.
       </ErrorText>
     )
   } else if (channelInfo?.private && !isUserInChannel) {
-    subtitle = <ErrorText>This channel is private and requires an invite to join.</ErrorText>
+    descriptionText = <ErrorText>This channel is private and requires an invite to join.</ErrorText>
```

Comment:
I'm not sure that either of these two "Errors" really make sense to display in place of the description? It seems like they should just change the Card action (and maybe this text would display in a tooltip for the action button? unsure)

Reply:
Yeah I wasn't sure either. For this error specifically, I thought it made sense to display this error in place of the description, since we don't want to display descriptions of private channels, I think? And displaying the default "No channel description" text seemed a bit weird. This is how the card looks like currently for private channels:

![image](https://user-images.githubusercontent.com/4757944/226172740-111b9fab-011a-444c-a70e-720f7a8d3c1c.png)

For the banned error message, I first display the channel info card normally (which means banned users will be able to see channel banner/badge/description/user count), and when they click the "Join" button the card displays the error message:

![image](https://user-images.githubusercontent.com/4757944/226172972-3ecd66aa-7486-4660-84ba-286b22afbaa0.png)

But yeah, now that I think about it, this might not be the best way to deal with that. I think it makes sense to think about which channel information we want to show to the user when they first open the card (for private channels that's none?), and then how to display the error message (for banned users they only realized they're banned _after_ they clicked the "Join" button).

Reply:
I think if you're not showing descriptions of private channels, then you should have some kind of content there saying it's private. It's not an error that it's private, and thus treating it like an error makes no sense. A better thing you have would have like a lock icon and text saying it's private. In general, don't use error-colored text treatments for things that are not errors.

As far as errors for channels you are banned from, I think replacing the description is weird, and it would be better if that error goes somewhere else (honestly it may make more sense as a dialog, using this popover as a modal for joining channels and surface for displaying the outcome of that feels incredibly weird to me)

Reply:
Related to that last point about it being weird to use this as the surface for displaying results of this operation, handling the navigation in `onJoinClick`'s success handler doesn't really work. That should really be handled via the action creator itself, and you should treat any callbacks you have on `abortableThunk`s as valid only as long as the component is mounted. We want to navigate to the newly joined channel whether or not this popover is still up, so it doesn't make sense to do that behavior (or display errors, or whatever else) inside the callback from the `abortableThunk`.